aten/src/ATen/native/cuda/CUDALoops.cuh

```diff
@@ -129,17 +129,22 @@ static inline void launch_unrolled_kernel(int64_t N, const func_t& f, array_t da
 }
 
 template<int nt, int vt, typename func_t>
-C10_LAUNCH_BOUNDS_2(nt, 4)
+C10_LAUNCH_BOUNDS_2(nt, 2)
 __global__ void elementwise_kernel(int N, func_t f) {
-  int tid = threadIdx.x;
-  int nv = nt * vt;
-  int idx = nv * blockIdx.x + tid;
+  // vt is elements unroll per thread
+  int start_idx = vt * (threadIdx.x + blockIdx.x * blockDim.x);
+  int grid_stride = vt * (gridDim.x * blockDim.x);
+
   #pragma unroll
-  for (int i = 0; i < vt; i++) {
-    if (idx < N) {
-      f(idx);
-      idx += nt;
-    }
+  for (int i = start_idx; i < N; i += grid_stride) {
```

Comment:
Shouldn't this be `i + vt < N` ?

Reply:
Updated with new commit, I didn't see any performance degradation because of the additional check.